src/lib/core/cbus.c

```diff
@@ -177,6 +187,145 @@ cpipe_destroy(struct cpipe *pipe)
 	TRASH(pipe);
 }
 
+struct lcpipe *
+lcpipe_new(const char *consumer)
+{
+	struct lcpipe *pipe = xmalloc(sizeof(*pipe));
+
+	stailq_create(&pipe->input);
+
+	pipe->n_input = 0;
+	pipe->max_input = INT_MAX;
+
+	acquire_consumer(&pipe->endpoint, consumer);
+
+	return pipe;
+}
+
+/**
+ * Flush all staged messages into consumer output and wake up it after.
+ */
+static void
+lcpipe_flush(struct lcpipe *pipe)
+{
```

Comment:
This function is a line to line copy-paste of `cpipe_flush_cb`.

Reply:
it's true, im move all duplicated code in `move_messages` function